Foundation/JSONSerialization.swift

```diff
@@ -462,15 +433,30 @@ private struct JSONWriter {
     }
 
     mutating func serializeNumber(_ num: NSNumber) throws {
-        if num.doubleValue.isInfinite || num.doubleValue.isNaN {
-            throw NSError(domain: NSCocoaErrorDomain, code: CocoaError.propertyListReadCorrupt.rawValue, userInfo: ["NSDebugDescription" : "Number cannot be infinity or NaN"])
-        }
-        
-        switch num._cfTypeID {
-        case CFBooleanGetTypeID():
-            serializeBool(num.boolValue)
-        default:
-            writer(_serializationString(for: num))
+        if CFNumberIsFloatType(num._cfObject) {
+            let dv = num.doubleValue
+            if !dv.isFinite {
+                let value: String
+                if dv.isNaN {
+                    value = "NaN"
```

Comment:
This is inconsistent with `NSJSONSerialization`, which does not support `NaN` or infinite values. This should arguably `preconditionFailure`/`fatalError`.

Reply:
(Seeing also as we claim in `isValidJSONValue` that infinite numbers/`NaN` are not valid, we shouldn't support them here.)

Reply:
This is actually to exactly match the error messaging on Darwin, not to serialise the value. `value` is used in the `NSError` below.

Reply:
You're totally right, of course — I'd misread this. 😬